### PR TITLE
Add test that validates attach partition fails if structure differs because of materialized column

### PR DESCRIPTION
**Pull request by arthurpassos**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add test that validates attach partition fails if structure differs because of materialized column

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit f04debe6484fce2b8d8529a81d4d1bb81dc012fd with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60418/f04debe6484fce2b8d8529a81d4d1bb81dc012fd/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60418/f04debe6484fce2b8d8529a81d4d1bb81dc012fd/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: tests/queries/0_stateless/02888_attach_partition_from_different_tables.sql [hunk 1]
```diff
@@ -88,3 +88,25 @@ ALTER TABLE attach_partition_t6 ATTACH PARTITION tuple() FROM attach_partition_t
...
+insert into attach_partition_t7 values (1, 2);
+
+CREATE TABLE attach_partition_t8 (
+                                     a UInt32,
+                                     b UInt32
+)
+    ENGINE = MergeTree
+PARTITION BY a ORDER BY a;
+
+ALTER TABLE attach_partition_t8 ATTACH PARTITION ID '1' FROM attach_partition_t7; -- {serverError 122};
```
Comment (Algunenano) [inline tests/queries/0_stateless/02888_attach_partition_from_different_tables.sql:112]:
```suggestion
ALTER TABLE attach_partition_t8 ATTACH PARTITION ID '1' FROM attach_partition_t7; -- {serverError INCOMPATIBLE_COLUMNS};
```

**Review by Algunenano — CHANGES_REQUESTED** [review]
Please create a new file to add new tests. It makes it simpler to track and test them individually.

File: tests/queries/0_stateless/02888_attach_partition_from_different_tables.sql (hunk 1 above)
Comment (arthurpassos) [reply tests/queries/0_stateless/02888_attach_partition_from_different_tables.sql:112]: Done

**Review by Algunenano — APPROVED** [review]

**Pull request merged**